src/components/LoginSignUp/AuthenticationService.js

```diff
@@ -0,0 +1,292 @@
+// @flow
+import { create } from "apisauce";
+import { version } from "../../../package.json";
+import Config from "react-native-config";
+import SInfo from "react-native-sensitive-info";
+import * as RNLocalize from "react-native-localize";
+import RNSInfo from "react-native-sensitive-info";
+import { sign } from "react-native-pure-jwt";
+
+const HOST = "https://www.inaturalist.org";
+
+// User agent being used, when calling the iNat APIs
+const USER_AGENT = `iNaturalist/${version} (ReactNative)`;
+
+const JWT_TOKEN_EXPIRATION_MINS = 25; // JWT Tokens expire after 30 mins - consider 25 mins as the max time (safe margin)
+
+class AuthenticationService {
+  /**
+   * Creates base API client for all requests
+   * @param additionalHeaders any additional headers that will be passed to the API
+   */
+  static createAPI( additionalHeaders: any ): any {
+    return create( {
+      baseURL: HOST,
+      headers: { "User-Agent": USER_AGENT, ...additionalHeaders }
+    } );
+  }
+
+  /**
+   * Returns the API access token to be used with all iNaturalist API calls
+   *
+   * @param useJWT if true, we'll use JSON Web Token instead of the "regular" access token
+   * @param allowAnonymousJWTToken (optional=false) if true and user is not logged-in, use anonymous JWT
+   * @returns {Promise<string|*>} access token, null if not logged in
+   */
+  static async getAPIToken( useJWT: boolean = false,  allowAnonymousJWTToken: boolean = false ): Promise<?string> {
+    let isLoggedIn = await this.isLoggedIn();
+    if ( !isLoggedIn ) {
+      return null;
+    }
+
+    if ( useJWT ) {
+      return this.getJWTToken( allowAnonymousJWTToken );
+    } else {
+      const accessToken = await RNSInfo.getItem( "accessToken" );
+      return `Bearer ${accessToken}`;
+    }
+  }
+
+  /**
+   * Returns the access token to be used in case of an anonymous JWT (e.g. used when getting taxon suggestions)
+   * @returns {Promise<string>}
+   */
+  static async getAnonymousJWTToken(): string {
+    const claims = {
+      application: "android",
```

Comment:
Should we make this platform specific and check whether a user is running android or ios here instead of assuming android?

Reply:
Fixed